src/perturbIC.py

```diff
@@ -1,52 +1,228 @@
-
 # Martin Dix martin.dix@csiro.au
 
+import os
 import argparse
-import umfile
-from um_fileheaders import *
-from numpy.random import MT19937, RandomState, SeedSequence
-
-parser = argparse.ArgumentParser(description="Perturb UM initial dump")
-parser.add_argument('-a', dest='amplitude', type=float, default=0.01,
-                    help = 'Amplitude of perturbation')
-parser.add_argument('-s', dest='seed', type=int, required=True,
-    help = 'Random number seed (must be non-negative integer)')
-parser.add_argument('ifile', help='Input file (modified in place)')
-
-args = parser.parse_args()
-
-if args.seed >= 0:
-    rs = RandomState(MT19937(SeedSequence(args.seed)))
-else:
-    raise Exception('Seed must be positive')
-
-f = umfile.UMFile(args.ifile, 'r+')
-
-# Set up theta perturbation.
-nlon = f.inthead[IC_XLen]
-nlat = f.inthead[IC_YLen]
-# Same at each level so as not to upset vertical stability
-perturb = args.amplitude * (2.*rs.random(nlon*nlat).reshape((nlat,nlon)) - 1.)
-# Set poles to zero (only necessary for ND grids, but doesn't hurt EG)
-perturb[0] = 0.
-perturb[-1] = 0.
-
-for k in range(f.fixhd[FH_LookupSize2]):
-    ilookup = f.ilookup[k]
-    lbegin = ilookup[LBEGIN] # lbegin is offset from start
-    if lbegin == -99:
-        break
-    # 4 is theta, 388 is thetavd (ENDGAME only)
-    if ilookup[ITEM_CODE] in (4, 388):
-        a = f.readfld(k)
-        # Note that using += ensures the datatype of a doesn't change
-        # (in case it's float32)
-        a += perturb
-        f.writefld(a,k)
-
-f.close()
+from numpy.random import PCG64, Generator
+import mule
+TIMESERIES_LBCODES = (31320, 31323)
+
+def parse_args():
+    """
+   Parse the command line arguments.
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    ----------
+    args_parsed : argparse.Namespace
+        Argparse namespace containing the parsed command line arguments.
+    """
+    parser = argparse.ArgumentParser(description="Perturb UM initial dump")
+    parser.add_argument('-a', dest='amplitude', type=float, default=0.01,
+                        help = 'Amplitude of the perturbation.')
+    parser.add_argument('-s','--seed', dest='seed', type=int,
+        help = 'The seed value used to generate the random perturbation (must be a non-negative integer).')
+    parser.add_argument('ifile', metavar="INPUT_PATH", help='Path to the input file.')
+    parser.add_argument('--validate', action='store_true',
+        help='Validate the output fields file using mule validation.')
+    args_parsed = parser.parse_args()
+    return args_parsed
+
+def create_random_generator(value=None):
+    """
+    Create the random generator object using the provided value as a seed.
+
+    Parameters
+    ----------
+    value : int
+           The seed value used to create the random generator.
+
+    Returns
+    ----------
+    numpy.random.Generator
+        The numpy random generator object.
+    """
+
+
+    if value < 0:
```

Comment:
```suggestion
    if value is not None and value < 0:
```
Otherwise if `value=None` it would fail because you cannot do `None < 0`